File: Atividade_8/Java/Xadrez.java
public class Xadrez {

    public class Peca {

        private String id;
        private String cor;

        public void setCor(String cor) {
            this.cor = cor;
        }

        public void setId(String id) {
            this.id = id;
        }

        public String getId() {
            return id;
        }

        public String getCor() {
            return cor;
        }

    }

    Peca[][] tabuleiro;

    public Xadrez(Peca[][] tabuleiro) {
        this.tabuleiro = tabuleiro;
    }

    public Xadrez() {
        this(new Peca[][] {});
    }

    public void moverPeca() {

    }

    public void capturarPeca(Peca inimigo, int linha, int coluna) {

    }

    public boolean podeMoverPara(Peca peca, int posicao) {
        return false;
    }

    public Peca getPeca(int linha, int coluna) {
        return null;
    }

}

File: Atividade_8/C/xadrez.c
typedef struct {
    char[255] tipo;
    char[255] cor;
} Peca;

typedef struct {
    Peca[8][8] tabuleiro;
} Xadrez;

void inicializarXadrez(Xadrez *xadrez, Peca[8][8] tabuleiro) {}
void inicializarPeca(Peca *peca, char[255] tipo, char[255] cor) {}

void moverPeca(Xadrez *xadrez, Peca *peca, int linha, int coluna) {}
void capturarPeca(Xadrez *xadrez, Peca *atual, Peca *inimiga) {}
int podeMover(Xadrez *xadrez, Peca *peca, int linha, int coluna) {}
char *cor(Peca *peca) {}
char *tipo(Peca *peca) {}